java/fury-core/src/main/java/org/apache/fury/Fury.java

```diff
@@ -1220,6 +1239,43 @@ public Object deserializeJavaObjectAndClass(FuryReadableChannel channel) {
     return deserializeJavaObjectAndClass(buf);
   }
 
+  @Override
+  public <T> T copy(T obj) {
+    if (obj == null) {
+      return null;
+    }
+    copyDepth++;
+    try {
+      return (T) classResolver.getOrUpdateClassInfo(obj.getClass()).getSerializer().copy(obj);
+    } catch (StackOverflowError e) {
```

Comment:
The try catch is expensive, the nested copy should not go to here. Maybe we should add a new copy method like `copyObject` which is used in nested copy, and replace all internal invoke of this `copy(T obj)` with `copyObject(T obj)`